index.js

```diff
@@ -0,0 +1,41 @@
+import Book from './modules/book.js';
+import spa from './modules/spa.js';
+import checkls from './modules/checklocalstorage.js';
+import { DateTime } from './modules/luxon.js';
+
+const awesomeBooks = document.getElementById('awesomeBooks');
+
+const library = new Book(awesomeBooks);
+
+document.querySelector('form').onsubmit = (e) => {
+  e.preventDefault();
+  const error = document.getElementById('error');
+  const { title, author } = e.target;
+  if (title.value.length < 3 || author.value.length < 3) {
+    error.innerHTML = 'input field should contain minimum of three characters!';
+    setTimeout(() => {
+      error.innerHTML = '';
+    }, 3000);
+  } else {
+    error.innerHTML = '';
+    library.add({
+      id: Date.now().toString(),
+      title: title.value,
+      author: author.value,
+    });
+    e.target.title.value = '';
+    e.target.author.value = '';
+  }
+};
+
+document.addEventListener('DOMContentLoaded', () => {
+  checkls(library);
+  spa();
+});
+
+const time = DateTime.now().toFormat("LLLL dd'th' tt");
+const date = document.querySelector('.date');
+
+date.innerHTML = time;
+
+library.books.forEach((book) => library.render(book));
```

Comment:
- [ ] Great job here 💯 but there is a small problem, when the page is reloaded the data will disappear, to fix this problem you have to get the data from the `localStorage` and render it 👍

> 💡 I can see that you already have a function called `checkls(library);` doing that, Great 🙌
> So you just have to render books inside the  `document.addEventListener('DOMContentLoaded', () => {})`
> For Example 💡
```js
document.addEventListener('DOMContentLoaded', () => {
  checkls(library);
  library.books.forEach((book) => library.render(book));
  spa();
});
```